[directxmath] simple main.cpp build error after adding #include <DirectXMath.h>

**Describe the bug**
I would like to use DirectXMath via vcpkg on Linux with external libs that uses `NULL`, but after I'm adding line:
`#include <DirectXMath.h>`
I got to error: expected expression

**Environment**
- OS: Ubuntu 24.04
- Compiler: tested gcc 14.2.0 and clang 19.1.1

**To Reproduce**
Steps to reproduce the behavior:
```
git clone --depth=1 -b test --recurse-submodules https://github.com/q4a/ci-vcpkg
mkdir ci-vcpkg/build && cd ci-vcpkg/vcpkg
./bootstrap-vcpkg.sh
cd ..
cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=./vcpkg/scripts/buildsystems/vcpkg.cmake
cmake --build build/
```

**Expected behavior**
Build without errors.

**Failure logs**
Local build:
```
$ cmake --build build/
[ 25%] Building CXX object CMakeFiles/main.dir/src/main.cpp.o
/home/test/git/ci-vcpkg/src/main.cpp:9:19: error: expected expression [clang-diagnostic-error]
    9 |   if (Str0 == NULL)
      |                   ^
14223 warnings and 1 error generated.
```
Error on github actions:
https://github.com/q4a/ci-vcpkg/actions/runs/13355754059/job/37298277092
github actions log archive: 
[logs_34452277127.zip](https://github.com/user-attachments/files/18815279/logs_34452277127.zip)

**Additional context**
@walbourn sorry for ping, but looks like there is simple way to fix this bug - just update:
https://github.com/microsoft/vcpkg/blob/d5ec528843d29e3a52d745a64b469f810b2cedbf/ports/directxmath/portfile.cmake#L24
from
`https://raw.githubusercontent.com/dotnet/runtime/v8.0.1/src/coreclr/pal/inc/rt/sal.h`
to
`https://raw.githubusercontent.com/dotnet/runtime/v9.0.2/src/coreclr/pal/inc/rt/sal.h`
(and SHA512 too). Can you do it?

--- a/ports/directxmath/portfile.cmake
+++ b/ports/directxmath/portfile.cmake
@@ -21,9 +21,9 @@ vcpkg_cmake_config_fixup(CONFIG_PATH share/directxmath)
 if(NOT VCPKG_TARGET_IS_WINDOWS)
     vcpkg_download_distfile(
         SAL_HEADER
-        URLS "https://raw.githubusercontent.com/dotnet/runtime/v8.0.1/src/coreclr/pal/inc/rt/sal.h"
+        URLS "https://raw.githubusercontent.com/dotnet/runtime/v9.0.2/src/coreclr/pal/inc/rt/sal.h"
         FILENAME "sal.h"
-        SHA512 0f5a80b97564217db2ba3e4624cc9eb308e19cc9911dae21d983c4ab37003f4756473297ba81b386c498514cedc1ef5a3553d7002edc09aeb6a1335df973095f
+        SHA512 8085f67bfa4ce01ae89461cadf72454a9552fde3f08b2dcc3de36b9830e29ce7a6192800f8a5cb2a66af9637be0017e85719826a4cfdade508ae97f319e0ee8e
     )
 
     file(INSTALL

--- a/ports/directxmath/vcpkg.json
+++ b/ports/directxmath/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "directxmath",
   "version-date": "2024-12-02",
+  "port-version": 1,
   "description": "DirectXMath SIMD C++ math library",
   "homepage": "https://github.com/Microsoft/DirectXMath",
   "documentation": "https://docs.microsoft.com/windows/win32/dxmath/directxmath-portal",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2306,7 +2306,7 @@
     },
     "directxmath": {
       "baseline": "2024-12-02",
-      "port-version": 0
+      "port-version": 1
     },
     "directxmesh": {
       "baseline": "2024-10-28",

--- a/versions/d-/directxmath.json
+++ b/versions/d-/directxmath.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c93ddde0a2ee30591dc28b758308ddb4bef60156",
+      "version-date": "2024-12-02",
+      "port-version": 1
+    },
+    {
       "git-tree": "bf54dfb76f186cf9b58dfe7d06150667ec54babc",
       "version-date": "2024-12-02",
       "port-version": 0

